src/flint/fq_default_extended.jl

```diff
@@ -533,11 +533,14 @@ function _fq_field_from_fmpz_mod_poly_in_disguise(f::FqPolyRingElem, s)
   # It is in K but the first 4 bytes are the type
   # Temporary hack
   #_K = _get_raw_type(FpField, K)
-  ccall((:fq_default_ctx_init_modulus, libflint), Nothing,
-        (Ref{FqField}, Ref{FqPolyRingElem}, Ptr{Nothing}, Ptr{UInt8}),
-        #z, f, _K.ninv, string(s))
-        z, f, pointer_from_objref(K) + 2 * sizeof(Cint), string(s))
-  finalizer(_FqDefaultFiniteField_clear_fn, z)
+  ss = string(s)
```

Comment:
```suggestion
  ss = string(s)::String
```
or else give `s` a type to ensure type stability and avoid invalidations?

Reply:
I don't understand. The type of `s` will always be known at compile time (it will be a `Symbol`).

Reply:
Sorry for my indeed very confusingly worded comment, and "type stability" definitely was the wrong expression.

Thing is, statically and so e.g. for precompilation, the compiler has no information about `s` (it is declared as `Any`), and it has to rely on inference for `string(::Any)`, but that is not enough to deduce that the return type is always a string:
```
julia> f(x::Vector{Any}) = string(x[1])
f (generic function with 1 method)

julia> @code_warntype f(Any[1])
MethodInstance for f(::Vector{Any})
  from f(x::Vector{Any}) @ Main REPL[3]:1
Arguments
  #self#::Core.Const(f)
  x::Vector{Any}
Body::Any
1 ─ %1 = Base.getindex(x, 1)::Any
│   %2 = Main.string(%1)::Any
└──      return %2
```
Note the `::Any` annotation for the return type.

As a result, it is forced to compile this code at runtime and cannot make use of code caching

If either the argument signature is narrowed to `s::Symbol` (assuming that covers all cases?), or a type annotation is added, this should go away (and hopefully fix the failing "invalidation" CI job)